lib/active_merchant/billing/gateways/visanet_peru.rb

```diff
@@ -88,10 +84,12 @@ def scrub(transcript)
 
       def add_invoice(params, money, options)
         # Visanet Peru expects a 9-digit numeric purchaseNumber
-        purchase_number = options[:purchase_number] || rand(100000000 .. 1000000000).to_s
-        params[:purchaseNumber] = purchase_number
-        params[:externalTransactionId] = purchase_number
-        params[:merchantId] = options[:merchant_id]
+        regex_purchase_number = /^[0-9]{9}$/
+        if options[:order_id] && options[:order_id].to_s =~ regex_purchase_number
+          params[:purchaseNumber] = options[:order_id].to_s
+        else
+          params[:purchaseNumber] = (SecureRandom.random_number(900_000_000) + 100_000_000).to_s
+        end
```

Comment:
@rwdaigle 

accepted. `externalTransactionId` was used in the earlier version. got rid of it thinking `order_id` being merchant supplied, will there be value in sending it back along with the response. I see now it will serve good purpose as a response identifier for the merchant. will make both changes.